### PR TITLE
ISPN-8997 Deploy artifacts

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-8997

### CODE REVIEW

**Comment by slaskawi** [comment]
@tristantarrant Ready for review.

File: Jenkinsfile [hunk 1]
```diff
@@ -70,6 +70,20 @@ pipeline {
                 sh 'find . -name "hs_err_*" -exec echo {} \\; -exec grep "^# " {} \\;'
             }
         }
+
+        stage('Deploy') {
+            when {
+                allOf {
+                    branch 'master'
+                    equals expected: 0, actual: currentBuild.rawBuild.getAction(AbstractTestResultAction.class).failCount
```
Comment (danberindei) [inline Jenkinsfile]: I think this is way too specific, the build could fail before it tries to run any test.
Why not use a `failure` block like https://github.com/infinispan/infinispan/pull/5897/files?

Comment (slaskawi) [reply Jenkinsfile]: I agree. @rigazilla did a pretty nice job by extracting `failure` and `success` scenarios in https://github.com/infinispan/infinispan/pull/5897. I guess I should put this there...

File: Jenkinsfile [hunk 2]
```diff
@@ -125,6 +125,11 @@ pipeline {
                         body: '${DEFAULT_CONTENT}'
                     }
                 }
+                if (!env.BRANCH_NAME.startsWith('PR-')) {
+                    configFileProvider([configFile(fileId: 'maven-settings-with-deploy-snapshot', variable: 'MAVEN_SETTINGS')]) {
+                        sh "$MAVEN_HOME/bin/mvn deploy -B -V -e -s $MAVEN_SETTINGS -Dmaven.test.failure.ignore=true -Dansi.strip -DskipTests=true"
```
Comment (danberindei) [inline Jenkinsfile]: I suggest using `mvn deploy:deploy` so that we don't compile or run the tests again.

**Comment by rvansa** [comment]
@slaskawi Is this PR still valid?

**Comment by danberindei** [comment]
@rvansa I was hoping to get a build without failures to actually test the deployment before integrating, but that's still pretty hard to obtain. I triggered one more build today, but I'll integrate it regardless of the result and wait for a green build on master to test it.

**Comment by danberindei** [comment]
The last build had some test failures as well, so I integrated it and we'll check for problems in the master builds :)

**Pull request merged**